amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/bulk/hold/DefaultHoldBulkMonitor.java

```diff
@@ -69,18 +72,64 @@ public HoldBulkStatus getBulkStatus(String bulkStatusId)
     }
 
     @Override
-    public List<HoldBulkStatus> getBulkStatusesForHold(String holdId)
+    public void cancelBulkOperation(String bulkStatusId, String reason)
     {
-        return Optional.ofNullable(holdProcessRegistry.get(holdId))
-            .map(bulkProcessDetailsList -> bulkProcessDetailsList.stream()
-                .map(HoldBulkProcessDetails::bulkStatusId)
-                .map(this::getBulkStatus)
-                .filter(Objects::nonNull)
-                .sorted(Comparator.comparing(HoldBulkStatus::endTime, Comparator.nullsLast(Comparator.naturalOrder()))
-                    .thenComparing(HoldBulkStatus::startTime, Comparator.nullsLast(Comparator.naturalOrder()))
-                    .reversed())
-                .toList())
-            .orElse(Collections.emptyList());
+        bulkCancellationsCache.put(bulkStatusId, reason);
```

Comment:
This + related cache implementation seems to me to imply that we're only ever going to store the "cancellationReason" and nothing else? What if in few months from now we decide to add some more fields to the cancellation request? I would have thought we'd store a serializable version of the cancellation request, that would be way more flexible and easier to manage in case the request ever changes in the future?

Reply:
Reimplemented the cancellation cache to store the `BulkCancellationRequest` object instead of the "cancellationReason".